src/unix/fs.c

```diff
@@ -1310,6 +1311,29 @@ static ssize_t uv__fs_copyfile(uv_fs_t* req) {
     }
   }
 
+  /**
+   * Change the timestamps of the destination file to match the source file.
+   */
+#if defined(__APPLE__)
+  times[0] = src_statsbuf.st_atimespec;
+  times[1] = src_statsbuf.st_mtimespec;
+#else
+  times[0] = src_statsbuf.st_atim;
+  times[1] = src_statsbuf.st_mtim;
+#endif
+
+  if (utimensat(AT_FDCWD, req->new_path, times, 0) == -1) {
+    err = UV__ERR(errno);
+    goto out;
+  }
+
+  /*
+   * Change the ownership and permissions of the destination file to match the
+   * source file.
+   * `cp -p` does not care about errors here, so we don't either.
+   */
+  fchown(req->file, src_statsbuf.st_uid, src_statsbuf.st_gid);
```

Comment:
Shouldn't we be using `dstfd` here?
```suggestion
  fchown(dstfd, src_statsbuf.st_uid, src_statsbuf.st_gid);
```

Reply:
Wow, nice catch, thanks!

Reply:
Fixed in https://github.com/libuv/libuv/commit/b1db4a1b7f326df095501fc63c9ae94d7ae1f3c7